app/uk/gov/hmrc/tai/model/domain/income/TaxCodeIncome.scala

```diff
@@ -45,10 +46,18 @@ case object PotentiallyCeased extends TaxCodeIncomeSourceStatus
 case object Ceased extends TaxCodeIncomeSourceStatus
 
 object TaxCodeIncomeSourceStatus{
-  implicit val formatTaxCodeIncomeSourceStatus = new Format[TaxCodeIncomeSourceStatus] {
-    override def reads(json: JsValue): JsSuccess[TaxCodeIncomeSourceStatus] = ???
+  implicit val formatTaxCodeIncomeSourceStatus: Format[TaxCodeIncomeSourceStatus] = new Format[TaxCodeIncomeSourceStatus] {
+    override def reads(json: JsValue): JsSuccess[TaxCodeIncomeSourceStatus] = JsSuccess(formatTaxCodeIncomeSourceStatusMap.getOrElse(json.as[String],
+      throw new IllegalArgumentException("Invalid Tax component type")))
     override def writes(taxCodeIncomeSourceStatus: TaxCodeIncomeSourceStatus) = JsString(taxCodeIncomeSourceStatus.toString)
   }
+
+  private val formatTaxCodeIncomeSourceStatusMap: Map[String, TaxCodeIncomeSourceStatus] = Map(
```

Comment:
TaxCodeIncome already contains live/ceased logic, is this duplicated somewhere? Again don't need this map just pass in the TaxCodeIncomeSourceStatus as itself and not a string